app/assets/javascripts/discourse/tests/acceptance/topic-edit-timer-test.js

```diff
@@ -183,6 +211,21 @@ acceptance("Topic - Edit timer", function (needs) {
     assert.ok(regex.test(text));
   });
 
+  test("schedule publish to category - does not show for a public topic", async function (assert) {
+    updateCurrentUser({ moderator: true });
+    const timerType = selectKit(".select-kit.timer-type");
+
+    await visit("/t/internationalization-localization");
+    await click(".toggle-admin-menu");
+    await click(".admin-topic-timer-update button");
+
+    await timerType.expand();
+    assert.notOk(
+      timerType.rowByValue("publish_to_category").exists(),
+      "publish to category is not allowed for public topics (not PM or private category)"
```

Comment:
Minor but since this is UI related, I think it'll be better if the description of the assertion is something like "publish to category is not displayed on public topics". Technically, permissions are all handled server side.